topic-operator/src/main/java/io/strimzi/operator/topic/Session.java

```diff
@@ -154,6 +155,35 @@ public void start(Promise<Void> start) {
             adminClientProps.setProperty(SslConfigs.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_CONFIG, "HTTPS");
         }
 
+        if (Boolean.valueOf(config.get(Config.TLS_SASL_ENABLED))) {
+            adminClientProps.setProperty(AdminClientConfig.SECURITY_PROTOCOL_CONFIG, config.get(Config.TLS_SECURITY_PROTOCOL));
+            adminClientProps.setProperty(SslConfigs.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_CONFIG, "");
+
+            adminClientProps.setProperty(SslConfigs.SSL_KEY_PASSWORD_CONFIG, config.get(Config.TLS_KEY_PASSWORD));
+            adminClientProps.setProperty(SaslConfigs.SASL_MECHANISM, config.get(Config.TLS_SASL_MECHANISM));
+
+            switch (config.get(Config.TLS_SASL_MECHANISM)) {
+                case "PLAIN": {
+                    String jaasTemplate = "org.apache.kafka.common.security.plain.PlainLoginModule required username=\"%s\" password=\"%s\";";
+                    String jaasCfg = String.format(jaasTemplate, config.get(Config.TLS_SASL_JAAS_USERNAME), config.get(Config.TLS_SASL_JAAS_PASSWORD));
+
+                    adminClientProps.setProperty(SaslConfigs.SASL_JAAS_CONFIG, jaasCfg);
+                }
+                break;
+                case "SCRAM-SHA-512": {
+                    String jaasTemplate = "org.apache.kafka.common.security.scram.ScramLoginModule required username=\"%s\" password=\"%s\";";
+                    String jaasCfg = String.format(jaasTemplate, config.get(Config.TLS_SASL_JAAS_USERNAME), config.get(Config.TLS_SASL_JAAS_PASSWORD));
+
+                    adminClientProps.setProperty(SaslConfigs.SASL_JAAS_CONFIG, jaasCfg);
+                }
+                break;
+                default:
+                    String jaasCfg = "";
+                    throw new IllegalArgumentException(
+                            "Undefined frame number: " + config.get(Config.TLS_SASL_MECHANISM));
```

Comment:
I'm not sure I undertand the message in the exception. Shouldn't it say something like _SASL is enabled by not SASL mechanism is specified_?